### ISSUE TITLE
User doesn't persist

### ISSUE THREAD

**Issue by thomasorten** [sidferreira/aor-firebase-client#26]
I can't get the user to persist. 

I can login fine, but upon browser refresh, firebase.auth().currentUser is null. 
The token firebaseToken is set in localStorage.

Everything else is pretty similar to your demo.

And, how is data structured in firebase? E.g. from your demo?

**Comment by sidferreira** [comment]
Noticed it too, but I'm wondering if this is a change in Firease's API.

I'm preparing a new version, with a lot of breaking changes, which will allow any schema. Will add more examples too.

**Comment by ejferg** [comment]
@sidferreira The AUTH_LOGOUT action is being triggered when you refresh the page. Which logs the user out causing currentUser to be null.

**Comment by ejferg** [comment]
@sidferreira @thomasorten This is how I solved the problem.

https://gist.github.com/ejferg/c025e202868da4761e737446166579c6

**Comment by sidferreira** [comment]
@ejferg Thanks a lot man! I think today I'll be able to check this and other stuff and update the tool.

**Comment by afilp** [comment]
@sidferreira I noticed that this worked for the 1st refresh but not for the 2nd, since `auth` is `null`.

**Comment by sidferreira** [comment]
will take a look!

**Comment by afilp** [comment]
@sidferreira I think this may have been my mistake, just mentioning it so that you will not lose time.

**Comment by afilp** [comment]
I think I found something. It has to do with the timeout in this line:

https://github.com/sidferreira/aor-firebase-client/blob/d58d737f718944f3e3c30c7ef40813de7ec0d363/src/authClient.js#L42

`        const timeout = (!firebase.auth().currentUser && localStorage.getItem(config.localStorageTokenName)) ? 1000 : 100`

It appears that if I increase the `1000` value to `4000` then I am not forced to logout.

Which means that `hot reloading` needs a little more time than the normal `production` run?

I did the following change for DEV only and it works (but I feel that it is just a hacky way):

`        const timeout = (!firebase.auth().currentUser && localStorage.getItem(config.localStorageTokenName)) ? (process.env.NODE_ENV !== 'production' ? 4000 : 1000) : 100`
